packager/hls/base/media_playlist.cc

```diff
@@ -599,6 +604,11 @@ void MediaPlaylist::AddSegmentInfoEntry(const std::string& segment_file_name,
           << start_time << ".";
       entries_.emplace_back(new DiscontinuityEntry());
     }
+  } else if (hls_params_.media_sequence_number > 0 &&
+    hls_params_.media_sequence_number == media_sequence_number_) {
+    // When there's a forced media_sequence_number, start with discontinuity.
```

Comment:
Yes, and this may actually be important.

Once a restart happens, we're kinda tricking the player here with the media sequence number: some later playlist refresh will get incremental sequence numbered segments, and so the player may understand it as a continuation. But that doesn't mean there's not a discontinuity here: segments were lost, and the packager was restarted (which I understand changes the point where segments are sliced and so changes the PTS distances, generating a gap for the player). So, if we put a discontinuity tag there, it should be easier for players to keep playing. 

But that's just my reasoning. I don't remember right now if it was actually REQUIRED for the players to keep on playing in my tests, but I understand it as a DESIRED and CORRECT logic anyways. The bogus point in this is when you start the stream for the first time, which is NOT a discontinuity, and yet the tag is put there. But given that this does not breaks anything, I don't consider it a big deal.